### ISSUE TITLE
Inconsistent LineLength behavior with URL-like strings

### ISSUE THREAD

**Issue by mvz** [rubocop/rubocop#9558]
When using double-quoted strings, a long URL-like string unexpectedly triggers a LineLength offense when it is used as argument to a method call using parentheses. This does not happen for single-quoted strings.

## Expected behavior

None of the following lines should trigger a LineLength offense:

```ruby
  foo = "http://www.example.com/foo/bar/baz/foo/bar/baz/foo/bar/baz/foo/bar/baz/foo/bar/baz/foo/bar/baz/foo/bar/baz/foo/bar/baz/'"
  bar("http://www.example.com/foo/bar/baz/foo/bar/baz/foo/bar/baz/foo/bar/baz/foo/bar/baz/foo/bar/baz/foo/bar/baz/foo/bar/baz/'")
  bar('http://www.example.com/foo/bar/baz/foo/bar/baz/foo/bar/baz/foo/bar/baz/foo/bar/baz/foo/bar/baz/foo/bar/baz/foo/bar/baz/')
  bar "http://www.example.com/foo/bar/baz/foo/bar/baz/foo/bar/baz/foo/bar/baz/foo/bar/baz/foo/bar/baz/foo/bar/baz/foo/bar/baz/'"
```

## Actual behavior

An LineLength offense is output for the line

```ruby
  bar("http://www.example.com/foo/bar/baz/foo/bar/baz/foo/bar/baz/foo/bar/baz/foo/bar/baz/foo/bar/baz/foo/bar/baz/foo/bar/baz/'")
```

## Steps to reproduce the problem

1. Create a file `foo.rb` containing
```ruby
# frozen_string_literal: true

def long
  foo = "http://www.example.com/foo/bar/baz/foo/bar/baz/foo/bar/baz/foo/bar/baz/foo/bar/baz/foo/bar/baz/foo/bar/baz/foo/bar/baz/'"
  bar("http://www.example.com/foo/bar/baz/foo/bar/baz/foo/bar/baz/foo/bar/baz/foo/bar/baz/foo/bar/baz/foo/bar/baz/foo/bar/baz/'")
  bar('http://www.example.com/foo/bar/baz/foo/bar/baz/foo/bar/baz/foo/bar/baz/foo/bar/baz/foo/bar/baz/foo/bar/baz/foo/bar/baz/')
  bar "http://www.example.com/foo/bar/baz/foo/bar/baz/foo/bar/baz/foo/bar/baz/foo/bar/baz/foo/bar/baz/foo/bar/baz/foo/bar/baz/'"
  foo
end
```
2. Create a file `.rubocop.yml` containing
```yaml
AllCops:
  NewCops: enable
```
3. Run `rubocop`

## RuboCop version

```
$ rubocop -V
1.11.0 (using Parser 3.0.0.0, rubocop-ast 1.4.1, running on ruby 3.0.0 x86_64-linux)
```
